### PR TITLE
refactor: use aerial to provide outline.

### PR DESCRIPTION
**Pull request by ayamir**
[Aerial.nvim](https://github.com/stevearc/aerial.nvim) has better outline support than [symbols-outline.nvim](https://github.com/simrat39/symbols-outline.nvim). #1058 will be solved.

### CODE REVIEW

**Comment by ayamir** [comment]
image

File: lua/modules/configs/completion/aerial.lua [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+		-- To see all available values, see :help SymbolKind
+		filter_kind = {
+			"Class",
+			"Constructor",
+			"Enum",
+			"Function",
+			"Interface",
+			"Module",
+			"Method",
+			"Struct",
```
Comment (CharlesChiuGit) [inline lua/modules/configs/completion/aerial.lua:110]: I think it's too few symbols for default usage, maybe add all symbols first?
```
Array
Boolean
Class
Constant
Constructor
Enum
EnumMember
Event
Field
File
Function
Interface
Key
Method
Module
Namespace
Null
Number
Object
Operator
Package
Property
String
Struct
TypeParameter
Variable
```

**Review by CharlesChiuGit — CHANGES_REQUESTED** [review]

File: lua/modules/configs/completion/aerial.lua (hunk 1 above)
Comment (CharlesChiuGit) [reply lua/modules/configs/completion/aerial.lua:110]: before
image

after
image

**Comment by CharlesChiuGit** [comment]
also the keymaps doesn't seem working.

**Comment by CharlesChiuGit** [comment]
also i think `aerial`'s telescope extension is so spot on that I think we should just change `go` to open the telescope preview.
How do u think? @ayamir @Jint-lzxy

**Comment by Jint-lzxy** [comment]
> also i think `aerial`'s telescope extension is so spot on that I think we should just change keymap:`go` to open it's telescope preview. How do u think?

Agree, but why not assign a keymap to each of them lol

**Comment by CharlesChiuGit** [comment]
> Agree, but why not assign a keymap to each of them lol

absolutely! that would be nice too!

**Review by CharlesChiuGit — APPROVED** [review]

**Comment by CharlesChiuGit** [comment]
I think it gd to go.

**Comment by Jint-lzxy** [comment]
Sorry I'm a bit busy with other stuff atm. I'll review this PR sometime tomorrow :D

**Review by CharlesChiuGit — APPROVED** [review]
LGTM

**Review by Jint-lzxy — APPROVED** [review]
LGTM 🍯 but we probably need to update the relevant parts of the wiki before merging this lol

**Pull request merged**